### PR TITLE
[2019.2] MacOS X Kitchen Salt Fixes

### PR DESCRIPTION
**Pull request by garethgreenaway**
### What does this PR do?
MacOS X Kitchen Salt Fixes

### What issues does this PR fix or reference?
N/A

### Tests written?
Yes.  Existing tests fixed.

### Commits signed with GPG?
Yes

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Review by waynew — APPROVED** [review]

File: tests/integration/states/test_pkg.py [hunk 1]
```diff
@@ -1059,6 +1059,7 @@ def test_pkg_cap_006_uptodate(self, grains=None):
             ret = self.run_state('pkg.removed', name=realpkg)
             self.assertSaltTrueReturn(ret)
 
+    @skipIf(True)
```
Comment (waynew) [inline tests/integration/states/test_pkg.py]: Is this failing on other OS? It looks like it should only be  running on redhat/debian :suspect:

Comment (garethgreenaway) [reply tests/integration/states/test_pkg.py]: Nope.  And it should be skipped, this was one of the spots where the tests were hanging.  The skipIf was just a test, not a permanent solution :smile:

Comment (waynew) [reply tests/integration/states/test_pkg.py]: That's freaky. I don't see anything in here that should be causing anything to hang - grains.get maybe? You could always test that with a `start = time.time()` and then `if time.time() - start > THRESHOLD: assert False, 'Took too long'`, or something to that effect. I haven't been able to get reliable output on my tests without raising an exception.

Comment (waynew) [reply tests/integration/states/test_pkg.py]: This test really would benefit from something like this though:

    @skipIf(not salt.utils.platform.is_darwin(), 'Test only available on macOS')

**Review by s0undt3ch — APPROVED** [review]

**Review by s0undt3ch — APPROVED** [review]

**Comment by waynew** [comment]
Looks like the current failure is due to another lib no longer supporting Python 3.4 - docker, at least

**Comment by s0undt3ch** [comment]
That issue will go away once 2019.2 is on nox :/

**Comment by waynew** [comment]
@s0undt3ch  Is there currently a merge forward/issue to track for that?

**Comment by s0undt3ch** [comment]
https://github.com/saltstack/salt/pull/52687

**Review by s0undt3ch — APPROVED** [review]

**Comment by s0undt3ch** [comment]
Now passing. Merging.

**Comment by s0undt3ch** [comment]
Should this go into 2019.2.1?

**Pull request merged**